x-pack/solutions/observability/test/functional/apps/dataset_quality/dataset_quality_details_failure_store.ts

```diff
@@ -0,0 +1,151 @@
+        createFailedLogRecord({
+          to: new Date().toISOString(),
+          count: 2,
+          dataset: nofailureStoreDatasetName,
+        }),
+        getLogsForDataset({
+          to: new Date().toISOString(),
+          count: 4,
+          dataset: nofailureStoreDatasetName,
+        }),
+      ]);
+    });
+
+    after(async () => {
+      await synthtrace.clean();
+      await synthtrace.deleteIndexTemplate(IndexTemplateName.Synht2);
+      await synthtrace.deleteComponentTemplate('synth.2@custom');
+      await synthtrace.deleteCustomPipeline('synth.2@pipeline');
+    });
+
+    it('should show "No failure store" card when failure store is disabled', async () => {
+      const { datasetQualityDetailsSummaryCardNoFailureStore } =
+        PageObjects.datasetQuality.testSubjectSelectors;
+
+      await PageObjects.datasetQuality.navigateToDetails({
+        dataStream: nofailureStoreDataStreamName,
+      });
+
+      await testSubjects.existOrFail(datasetQualityDetailsSummaryCardNoFailureStore);
+      const failedDocsCard = await testSubjects.getVisibleText(
+        datasetQualityDetailsSummaryCardNoFailureStore
+      );
+      expect(failedDocsCard).to.contain('No failure store');
+    });
+
+    it('should open failure store modal and save new config', async () => {
+      const {
+        datasetQualityDetailsSummaryCardFailedDocuments,
+        datasetQualityDetailsSummaryCardNoFailureStore,
+        datasetQualityDetailsEnableFailureStoreButton,
+        editFailureStoreModal,
+        failureStoreModalSaveButton,
+        enableFailureStoreToggle,
+      } = PageObjects.datasetQuality.testSubjectSelectors;
+
+      await PageObjects.datasetQuality.navigateToDetails({
+        dataStream: nofailureStoreDataStreamName,
+      });
+
+      await testSubjects.existOrFail(datasetQualityDetailsSummaryCardNoFailureStore);
+      await testSubjects.missingOrFail(datasetQualityDetailsSummaryCardFailedDocuments);
+
+      await testSubjects.click(datasetQualityDetailsEnableFailureStoreButton);
+
+      await testSubjects.existOrFail(editFailureStoreModal);
+
+      const saveModalButton = await testSubjects.find(failureStoreModalSaveButton);
+      expect(await saveModalButton.isEnabled()).to.be(false);
+
+      await testSubjects.click(enableFailureStoreToggle);
+
+      expect(await saveModalButton.isEnabled()).to.be(true);
+
+      await testSubjects.click(failureStoreModalSaveButton);
+
+      await testSubjects.missingOrFail(editFailureStoreModal);
+
+      await testSubjects.existOrFail(datasetQualityDetailsSummaryCardFailedDocuments);
+
+      await testSubjects.missingOrFail(datasetQualityDetailsSummaryCardNoFailureStore);
+
+      const failedDocsCard = await testSubjects.getVisibleText(
+        datasetQualityDetailsSummaryCardFailedDocuments
+      );
+      expect(failedDocsCard).to.contain('Failed documents');
+    });
+
+    it('should show failed docs count when failure store is enabled', async () => {
```

Comment:
Not sure if we should have this test like this, we would be relying on the previous one for this one to work right? This will become a flaky test if somehow the test are run in a different order, I would create a dedicated describe for this or maybe navigating to another dataset that has the failure store already enabled